### PR TITLE
Add libnpp recipe

### PR DESCRIPTION
**Pull request by adibbley**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

<hr>

xref: https://github.com/conda-forge/staged-recipes/issues/21382

### CODE REVIEW

File: recipes/libnpp/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+  - name: libnpp
+    files:
+      - lib/libnpp*.so.*                            # [linux]
+      - targets/{{ target_name }}/lib/libnpp*.so.*  # [linux]
+      - Library\bin\npp*.dll                        # [win]
+    requirements:
+      build:
+        - {{ compiler("c") }}
+        - {{ compiler("cxx") }}
+        - arm-variant * {{ arm_variant_type }}  # [aarch64]
```
Comment (robertmaynard) [inline recipes/libnpp/meta.yaml:48]:
```suggestion
        - arm-variant * {{ arm_variant_type }}  # [aarch64]
        - sysroot_{{ target_platform }} 2.17  # [linux]
```

File: recipes/libnpp/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+      - targets/{{ target_name }}/lib/libnpp*.so  # [linux]
+      - targets/{{ target_name }}/lib/stubs       # [linux]
+      - Library\include                           # [win]
+      - Library\lib                               # [win]
+    requirements:
+      run:
+        - libnpp >={{ version }}
+    requirements:
+      build:
+        - arm-variant * {{ arm_variant_type }}  # [aarch64]
```
Comment (robertmaynard) [inline recipes/libnpp/meta.yaml:83]:
```suggestion
        - {{ compiler("c") }}
        - {{ compiler("cxx") }}
        - arm-variant * {{ arm_variant_type }}  # [aarch64]
        - sysroot_{{ target_platform }} 2.17    # [linux]
```

File: recipes/libnpp/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    build:
+      skip: True  # [not linux]
+    requirements:
+      run:
+        - libnpp-dev >={{ version }}
+    files:
+      - targets/{{ target_name }}/lib/libnpp*_static.a
+    requirements:
+      build:
+        - arm-variant * {{ arm_variant_type }}  # [aarch64]
```
Comment (robertmaynard) [inline recipes/libnpp/meta.yaml:121]:
```suggestion
        - {{ compiler("c") }}
        - {{ compiler("cxx") }}
        - arm-variant * {{ arm_variant_type }}  # [aarch64]
        - sysroot_{{ target_platform }} 2.17    # [linux]
```

**Review by robertmaynard — CHANGES_REQUESTED** [review]
Feedback based on the meta.yaml for libcublas

**Review by bdice — APPROVED** [review]

**Review by robertmaynard — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks all! 🙏

Let's ship this one. Happy to follow up on anything in the feedstock

**Pull request merged**